modules/util/slice.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package util
+
+// IntersectString returns the intersection of the two string slices
+func IntersectString(a, b []string) []string {
```

Comment:
This may not be the best algorithm depending on how big these []string get, whether they can be or are sorted and how much of an intersection they have. 

If the slices are presorted you can generate an intersection in len(a)+len(b) time. 

The main question is how big a []string we're expecting here.